### PR TITLE
Adds export and runtime_deps support to scala rule

### PR DESCRIPTION
**Pull request by johnynek**
This brings the scala rules a step closer to feature parity with java.

### CODE REVIEW

File: tools/build_defs/scala/scala.bzl [hunk 1]
```diff
@@ -170,10 +193,19 @@ def _scala_macro_library_impl(ctx):
       scala = scalaattr,
       runfiles=runfiles)
 
+def _scala_library_impl(ctx):
+  return _lib(ctx, True)
+
+def _scala_macro_library_impl(ctx):
+  return _lib(ctx, False) # don't build the ijar for macros
```
Comment (damienmg) [inline tools/build_defs/scala/scala.bzl]: 2spaces

File: tools/build_defs/scala/scala.bzl [hunk 2]
```diff
@@ -170,10 +193,19 @@ def _scala_macro_library_impl(ctx):
...
+
 # Common code shared by all scala binary implementations.
 def _scala_binary_common(ctx, cjars, rjars):
   _write_manifest(ctx)
-  _compile(ctx, cjars, False)
+  _compile(ctx, cjars, False) # no need to build an ijar for an executable
```
Comment (damienmg) [inline tools/build_defs/scala/scala.bzl]: 2 spaces

**Comment by damienmg** [comment]
I don't really see how this exports attribute is different from a deps... Can you add a test that shows the difference?

**Comment by damienmg** [comment]
(Also, update the README.md?)

**Comment by johnynek** [comment]
Added the tests. I checked that they fail when the exports or runtime_deps are removed. It is not super clean to add negative tests now, is it? I would generate a bad build and show that it fails. Is that needed?

**Comment by damienmg** [comment]
Right it's not easy to have negative test that way (we do wrap in shell integration test for some case, but that is kind of heavyweight).

LGTM
Will merge tomorrow. time to sleep for me.

File: tools/build_defs/scala/scala.bzl [hunk 3]
```diff
@@ -117,14 +117,18 @@ cd $0.runfiles
       output=ctx.outputs.executable,
       content=content)
 
-def _collect_comp_run_jars(ctx):
-  compile_jars = set() # not transitive
-  runtime_jars = set() # this is transitive
-  for target in ctx.attr.deps:
+def _collect_jars(targets):
+  """Compute the runtime ant compile-time dependencies from the given targets
+  """
```
Comment (damienmg) [inline tools/build_defs/scala/scala.bzl]: nit: """ on the same line + ending .

**Comment by johnynek** [comment]
@damienmg thanks! Fixed the comments.

Really appreciate you quickly reviewing and merging!

**Comment by damienmg** [comment]
This was merged! Thanks for another contribution to Bazel!

**Pull request closed without merging**